Assets/Scripts/UI/SlipTimeUI.cs

```diff
@@ -9,41 +8,54 @@ public class SlipTimeUI : MonoBehaviour
     {
         public SlipTimeManager slipTimeManager;
 
-        public Text slipTimeUIText;
-
         private int slipTimeCharges;
 
+        // Fillable images for the sliptime meter and charge bars
+        public Image meter;
+        public Image[] bars;
+
         // Start is called before the first frame update
         private void Start()
         {
+            // Handle image setup
+            meter.type = Image.Type.Filled;
+            meter.fillAmount = 1;
+            foreach (var img in bars)
+            {
+                img.type = Image.Type.Filled;
+                img.fillAmount = 1;
+            }
             slipTimeCharges = slipTimeManager.slipTimeCharges;
-            slipTimeUIText.text = $"Slip Time Charges: {slipTimeCharges}";
         }
 
         // Update is called once per frame
-        private void Update()
+        private void LateUpdate()
         {
-            var slipTimeText = $"Slip Time Charges: {slipTimeManager.slipTimeCharges}";
-            
-            // Leave this code here for now. We may be able to re-use it later once we get a better UI for SlipTime
-            // if we don't have to constantly update the text box.
-            // if (slipTimeCharges != slipTimeManager.slipTimeCharges)
-            // {
-            //     slipTimeCharges = slipTimeManager.slipTimeCharges;
-            //     slipTimeUIText.text = $"Slip Time Charges: {slipTimeCharges}";
-            // }
+            if (slipTimeCharges != slipTimeManager.slipTimeCharges)
+            {
+                slipTimeCharges = slipTimeManager.slipTimeCharges;
+                // Reset bars to unfilled
+                foreach (var img in bars)
+                {
+                    img.fillAmount = 0;
+                }
+                // Fill only the bars we need
+                for(var i = 0; i<slipTimeCharges; i++)
+                {
+                    bars[i].fillAmount = 1;
+                }
+            }
 
+            // Sliptime meter shild get emptied when sliptime is engaged
```

Comment:
Typo: Sliptime meter **should** get emptied when sliptime is engaged

I noticed that the meter's fillAmount did not recharge when in slip-time. Was this intended? (I think I actually prefer it this way that we recharge while still in sliptime rather than reset as it allows more faster gameplay loop).

Reply:
I'm a bit confused here because it seems in conflict with the following comment -- sliptime *does* charge during sliptime. The idea is for this script to act as a pass-through for `SliptimeManager` as much as possible and match the state of the charge bars and the meter to the state of `SliptimeManager`. So the goal is that even if we change `SliptimeManager`'s mechanics, this script won't need to change.

Reply:
I think we are in agreement, see response below.